### PR TITLE
#28 Created JWT and RefreshToken

### PR DESCRIPTION
**Pull request by zharan96prog**

### CODE REVIEW

**Review by ICHTLAY — APPROVED** [review]

**Review by SashaKorniichuk — APPROVED** [review]

File: AccountService.Data/Entities/User.cs [hunk 1]
```diff
@@ -10,6 +12,8 @@ public class User : MongoIdentityUser<string>
         public string FirstName { get; set; }
         public string LastName { get; set; }
         public PersonalData UserData { get; set; }
+
```
Comment (pinkevmladchy) [inline AccountService.Data/Entities/User.cs:15]: delete empty line

File: AccountService.Domain/Services/ServiceAccount.cs [hunk 2]
```diff
@@ -56,22 +62,80 @@ public async Task RegisterUser(UserRegisterRequest userRequest)
...
+        {
+            var tokenHandler = new JwtSecurityTokenHandler();
+            var key = Encoding.UTF8.GetBytes(_configuration.GetValue<string>("Secret"));
+            var tokenDescriptor = new SecurityTokenDescriptor
+            {
+                Subject = new ClaimsIdentity(new Claim[]
+                {
+                    new Claim(ClaimTypes.Name, user.Id.ToString())
+                }),
+                Expires = DateTime.UtcNow.AddMinutes(1),
```
Comment (vladtymo) [inline AccountService.Domain/Services/ServiceAccount.cs]: Move token expires minutes value to config file

File: AccountService.WebApi/Controllers/AccountController.cs [hunk 3]
```diff
@@ -14,17 +16,39 @@ public AccountController(IServiceAccount serviceAccount)
...
+                SetRefreshTokenInCookie(response.RefreshToken);
+            return Ok(response);
+        }
+
+        private void SetRefreshTokenInCookie(string refreshToken)
+        {
+            var cookieOptions = new CookieOptions
+            {
+                HttpOnly = true,
+                Expires = DateTime.UtcNow.AddDays(7),
```
Comment (vladtymo) [inline AccountService.WebApi/Controllers/AccountController.cs]: Move minutes literal to config file

File: AccountService.Domain/Interfaces/IServiceAccount.cs [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
 namespace AccountService.Domain.Interfaces
 {
     public interface IServiceAccount
     {
         public Task RegisterUser(UserRegisterRequest user);
-        public Task LoginUser(UserLoginRequest user);
+        public Task<AuthenticateResponse> LoginUser(UserLoginRequest userRequest);
+        public Task<AuthenticateResponse> RefreshTokenAsync(string token);
+        public bool RevokeToken(string token);
+
```
Comment (vladtymo) [inline AccountService.Domain/Interfaces/IServiceAccount.cs]: Remove empty line

File: AccountService.Domain/RequestModels/RevokeTokenRequest.cs [hunk 5]
```diff
@@ -0,0 +1,8 @@
+﻿
```
Comment (vladtymo) [inline AccountService.Domain/RequestModels/RevokeTokenRequest.cs:1]: Remove empty line

**Review by vladtymo — CHANGES_REQUESTED** [review]

**Review by pinkevmladchy — APPROVED** [review]
good job

**Pull request merged**